### DISCUSSION TITLE
Game controller to dual joystick configuration

### DISCUSSION THREAD

**Discussion by mnemo70** [dirkwhoffmann/vAmiga#834] [Q&A] [answered]
Hi there,

I was trying to play the new dual-stick shooter Cecconoid on vAmiga and tried configuring a PS5 game controller (via USB).

My thought was to assign the left stick to joystick port 2 (movement) and the right stick to joystick port 1 (shooting). The ports are also selectable in the game options. But from what I see in the vAmiga UI you can only assign the whole controller once to either port 1 or 2.

So would it be possible to add this type of configuration or is it maybe already possible without me realizing? I have to admit, I did not understand the axis options in the controller configuration menu.

Thanks!

**Comment by mnemo70** [comment]
Here's a screenshot of the device config dialog.

<img width="569" alt="image" src="https://github.com/dirkwhoffmann/vAmiga/assets/30214405/d2fa3cb9-5837-4ec0-991b-09bbce90f2a9">

**Comment by dirkwhoffmann** [comment, marked as answer]
I don't plan to implement such a feature as the current SW architecture is incompatible with it. The whole HID code relies on one game controller being bound to one port.

**Reply by mnemo70** [reply]
Not what I wanted to hear, but thanks for the quick answer. :-)

**Comment by mnemo70** [comment]
Another idea: Would it at least be possible to emulate the CD32 gamepad which has 7 additional buttons apart from the directional ones? Cecconoid allows selecting the CD32 buttons to shoot and vAmiga could map the PS5 main buttons (square, circle, triangle and X) plus L/R1 or L/R2 and maybe Start to those seven buttons. This would also work on an Xbox controller, of course.

An explanation of the CD32 gamepad [can be found online](https://www.mrdictionary.net/PSCD32/diary/2019_08_09.htm).

This way there would be one controller mapped to one port.
